spec/jobs/checksum_report_job_spec.rb

```diff
@@ -18,45 +18,69 @@
-        expect(File).to exist(File.join(output_directory, Settings.checksum_report_job.csv_filename))
-        expect(bulk_action.druid_count_total).to eq(pids.length)
-        expect(bulk_action.druid_count_fail).to eq(0)
-        expect(bulk_action.druid_count_success).to eq(pids.length)
+      context 'happy path' do
+        before do
+          allow(Preservation::Client.objects).to receive(:checksums).with(druids: pids).and_return(csv_response)
+        end
+
+        it 'calls the presevation_catalog API, writes a CSV file, and records success counts' do
+          subject.perform(bulk_action.id,
+            output_directory: output_directory,
+            pids: pids,
+            groups: groups,
+            user: user)
+          expect(Preservation::Client.objects).to have_received(:checksums).with(druids: pids)
+          expect(File).to exist(File.join(output_directory, Settings.checksum_report_job.csv_filename))
+          expect(bulk_action.druid_count_total).to eq(pids.length)
+          expect(bulk_action.druid_count_fail).to eq(0)
+          expect(bulk_action.druid_count_success).to eq(pids.length)
+        end
+      end
+
+      context 'Preservation::Client throws error' do
+        before do
+          allow(Preservation::Client.objects).to receive(:checksums).with(druids: pids).and_raise(Preservation::Client::UnexpectedResponseError, 'ruh roh')
+        end
+
+        it 'calls the presevation_catalog API, does not write a CSV file, and records failure counts' do
+          subject.perform(bulk_action.id,
+            output_directory: output_directory_fail,
+            pids: pids,
+            groups: groups,
+            user: user)
+          expect(Preservation::Client.objects).to have_received(:checksums).with(druids: pids)
+          expect(File).not_to exist(File.join(output_directory_fail, Settings.checksum_report_job.csv_filename))
+          expect(bulk_action.druid_count_total).to eq(pids.length)
+          expect(bulk_action.druid_count_fail).to eq(pids.length)
+          expect(bulk_action.druid_count_success).to eq(0)
+        end
       end
     end
 
     context 'without authorization' do
       let(:ability) { instance_double(Ability, can?: false) }
 
+      before do
+        allow(Preservation::Client.objects).to receive(:checksums)
+      end
+
       it 'does not call the presevation_catalog API, does not write a CSV file, and records failure counts' do
-        allow(my_conn).to receive(:post)
-        subject.perform(bulk_action.id,
+        expect { subject.perform(bulk_action.id,
                         output_directory: output_directory_fail,
                         pids: pids,
                         groups: groups,
-                        user: user)
-        expect(my_conn).not_to have_received(:post).with('/objects/checksums', druids: pids, format: 'csv')
+                        user: user) }.to raise_error(RuntimeError, /ChecksumReportJob not authorized to view all content/)
```

Comment:
I'm assuming this is the ability filtering, so StandardError has to do with opening the output file?

Reply:
sounds reasonable